### PR TITLE
feat: add errors middleware

### PR DESCRIPTION
**Pull request by fjsandov**

### CODE REVIEW

File: src/app.js [hunk 1]
```diff
@@ -1,9 +1,14 @@
 const Koa = require('koa');
 const logger = require('koa-logger');
 const router = require('./routes');
+const config = require('./config');
+const { errorsMiddleware } = require('./middleware');
 
 const app = new Koa();
-app.use(logger());
+if (!config.env.isTest) {
```
Comment (negebauer) [inline src/app.js]: I think that in this types of `if` we have to be explicit with the condition.

```suggestion
if (config.env.isDev || config.env.isProd) {
```

File: src/middleware/errors.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+const { env } = require('../config');
+
+function logError(err) {
+  // TODO: config rollbar or something like that
+  if (!env.isTest) console.error(err); // eslint-disable-line no-console
```
Comment (negebauer) [inline src/middleware/errors.js]: Same as before comment.
```suggestion
  if (env.isDev || env.isProd) console.error(err); // eslint-disable-line no-console
```

File: src/test-utils/factories.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
+/* eslint-disable import/no-extraneous-dependencies */
+const { Factory } = require('rosie');
+
+Factory.define('koa-ctx').attrs({
```
Comment (negebauer) [inline src/test-utils/factories.js]: Multiple uses of the same string, move it to a variable. (can use `keymirror`)

File: src/middleware/errors.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  try {
+    await next();
+  } catch (err) {
+    ctx.status = err.status || 500;
+    if (ctx.status === 500) logError(err);
+    ctx.body = {
+      message:
+        ctx.status !== 500 && err.message
+          ? err.message
+          : ctx.t('errors.default'),
```
Comment (sasalatart) [inline src/middleware/errors.js]:
```suggestion
          : 'Server error',
```

^^ Remember that line assumes the app has i18n configured, which is not the case here (`t` is undefined)

File: src/middleware/errors.js (hunk 2 above)
Comment (negebauer) [reply src/middleware/errors.js]: I think you missed this one @fjsandov

**Pull request merged**